docs(google drive): add Google Drive OOTB connector doc

igpetrov opened the pull request:
closes #1161

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+
+## Prerequisites
+
+To start working with the **Google Drive connector**, a relevant OAuth token must be configured and stored as a secret in the user's cluster. The token has to have permission to read and create a folder and/or files from a desired Google Drive instance. Follow the steps from the appendix to find out more about creating an OAuth token and giving relevant permissions.
```
Let's link the `appendix&faq`

christinaausley replied:
Agreed here.

igpetrov replied:
Done

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+description: Create empty folders or documents from templates from your BPMN process.
+---
+
+The **Google Drive connector** allows you to create empty folders or documents from templates from your BPMN process.
```
Maybe a bit more verbose

```suggestion
The **Google Drive connector** allows you to create empty folders or documents on [Google Drive](https://drive.google.com/) from templates from your BPMN process.
```

MaxTru replied:
Is there a link we can add for `templates` (e.g., I have not used Google Drive templates) to get started with what they are?

igpetrov replied:
Added Google Drive link. For templates we don't have explicit reference, except one mentioned in FAQ section. Let me know if you think, we should add that one.

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+
+Currently, the **Google Drive connector** supports two flavors: create a new folder and create a file from a template.
+
+To use a **Google Drive connector** in your process, either change the type of existing task by clicking on it and using the wrench-shaped **Change type** context menu or create a new connector task by using the **Append connector** context menu. Follow our guide on using connectors to learn more.
```
> Follow our guide on using connectors to learn more.

Should be linked

igpetrov replied:
Done

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+2. In the **Select Operation** section, set the field value **Operation Type** as **Create Folder**.
+3. In the **Operation Details** section, set the field **Folder name** as the desired name of a folder you wish to create. For example, `MyNewFolder`. Alternatively, you could use a FEEL expression.
+4. _(optional)_ In the **Operation Details** section, set the field **Parent folder ID** to the desired parent, inside which a new folder will be created. Please, keep in mind, that if not specified, a new folder will be created in the Google Drive root folder of a user, who owns the OAuth token.
+5. _(optional)_ In the **Operation Details** section, set the field **Additional properties or metadata**, set Google Drive compatible properties or metadata. This property requires FEEL input. Check the appendix for known values and limitations.
```
Link to appendix

igpetrov replied:
Done

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+4. In the **Operation Details** section, set the field **Template ID** of the desired template.
+5. _(optional)_ In the **Operation Details** section, set the field **Parent folder ID** to the desired parent, inside which a new file will be created. Please, keep in mind, that if not specified, a new folder will be created in the Google Drive root folder of a user, who owns the OAuth token.
+6. In the **Operation Details** section, set the field **Template variables** as desired variables that will be applied to the template. The template variables a compatible with the Google Docs Requests API. This property requires FEEL input.
+7. _(optional)_ In the **Operation Details** section, set the field **Additional properties or metadata**, set Google Drive compatible properties or metadata. This property requires FEEL input. Check the appendix for known values and limitations.
```
Link to appendix

igpetrov replied:
Done

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+
+The **Google Drive connector** exposes Google Drive API response as a local variable called response.
+The following fields are available in the response variable:
+`googleDriveResourceId` - ID of a newly created resource
```
```suggestion
`googleDriveResourceId` - ID of the newly created resource
```

igpetrov replied:
Done

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+The **Google Drive connector** exposes Google Drive API response as a local variable called response.
+The following fields are available in the response variable:
+`googleDriveResourceId` - ID of a newly created resource
+`googleDriveResourceUrl` - human-readable URL of a newly created resource
```
```suggestion
`googleDriveResourceUrl` - human-readable URL of the newly created resource
```

igpetrov replied:
done

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
````diff
@@ -0,0 +1,200 @@
+1. Use **Result Variable** to store the response in a process variable. For example, `myResultVariable`.
+2. Use **Result Expression** to map fields from the response into process variables. For example:
+
+```
````
I don't get the example. The expression you put would map static fields into variables right?

So after execution the variable `googleDriveResourceId` would exist in my process with the value `1KMG4S_18kLNZa7ZCARtGIh8DmIoXXXXXXXXX`. Why would I do that?

Would a more appropriate example not be something like this:

```
{
  "newFolderID": response.googleDriveResourceId
}
```

MaxTru replied:
Or do I miss something?

igpetrov replied:
Yes, correct. Updated.

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+
+## How can I get Google Drive OAuth token?
+
+Refer to the official Google Drive API documentation page.
```
Link missing

igpetrov replied:
Rewrote the whole section

MaxTru reviewed (changes requested):
This connector already looks super nice!

I did a quick first review and found some things, primarily linking.

In general missing is update of `sidebars.js` (please also ensure your change works end-2-end by running `npm run start`)

On top (once we have this close to be ready for merge), we should also make the change in the `versioned_docs/version-8.0` (at least if we release this prior to October, which I assume). Otherwise the change would not be visible in the docs before the 8.1 release (in October)

christinaausley commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+
+## Create a Google Drive connector task
+
+Currently, the **Google Drive connector** supports two flavors: create a new folder and create a file from a template.
```
What do you mean by "flavors" here? Use cases? Instances?

igpetrov replied:
Two variants of templates (and use cases) - one to create a folder, one to create a file.

christinaausley replied:
Excellent! I would phrase it like this in the doc then.

"Currently, the **Google Drive connector** supports two types of templates: create a folder and create a file from a template."

igpetrov replied:
Updated

christinaausley commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+
+Currently, the **Google Drive connector** supports two flavors: create a new folder and create a file from a template.
+
+To use a **Google Drive connector** in your process, either change the type of existing task by clicking on it and using the wrench-shaped **Change type** context menu or create a new connector task by using the **Append connector** context menu. Follow our guide on using connectors to learn more.
```
When you say "Follow our guide on using connectors to learn more" -- please link to the connectors guide.

igpetrov replied:
Updated

christinaausley commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+2. In the **Select Operation** section, set the field value **Operation Type** as **Create Folder**.
+3. In the **Operation Details** section, set the field **Folder name** as the desired name of a folder you wish to create. For example, `MyNewFolder`. Alternatively, you could use a FEEL expression.
+4. _(optional)_ In the **Operation Details** section, set the field **Parent folder ID** to the desired parent, inside which a new folder will be created. Please, keep in mind, that if not specified, a new folder will be created in the Google Drive root folder of a user, who owns the OAuth token.
+5. _(optional)_ In the **Operation Details** section, set the field **Additional properties or metadata**, set Google Drive compatible properties or metadata. This property requires FEEL input. Check the appendix for known values and limitations.
```
"set the field **Additional properties or metadata** to..."?

I'm confused by "In the **Operation Details** section, set the field **Additional properties or metadata**, set Google Drive compatible properties or metadata" as it isn't a complete sentence. What do you mean here?

igpetrov replied:
What is meant here, there's a section called **Operation Details**. This section contains an input field **Additional properties or metadata**. In this section, user can input Google Drive compatible properties, e.g. _description_ of the folder.

christinaausley replied:
So do you mean...
"In the **Operation Details** section, you can set the **Additional properties or metadata** field to Google Drive compatible properties. For example, *description* of the folder."

igpetrov replied:
Yes, updated.

christinaausley commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+4. In the **Operation Details** section, set the field **Template ID** of the desired template.
+5. _(optional)_ In the **Operation Details** section, set the field **Parent folder ID** to the desired parent, inside which a new file will be created. Please, keep in mind, that if not specified, a new folder will be created in the Google Drive root folder of a user, who owns the OAuth token.
+6. In the **Operation Details** section, set the field **Template variables** as desired variables that will be applied to the template. The template variables a compatible with the Google Docs Requests API. This property requires FEEL input.
+7. _(optional)_ In the **Operation Details** section, set the field **Additional properties or metadata**, set Google Drive compatible properties or metadata. This property requires FEEL input. Check the appendix for known values and limitations.
```
Same comment here.

igpetrov replied:
Done

christinaausley commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+
+## How can I get Google Drive OAuth token?
+
+Refer to the official Google Drive API documentation page.
```
Please link to this 👍

igpetrov replied:
Rewrote the whole section

christinaausley commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,200 @@
+
+![Google Drive connector document template example](../img/connectors-googledrive-template-example.png)
+
+Now, in the Template variables field we can apply the following FEEL JSON object that has to be compatible with the Google Docs Requests API:
```
Is "Template variables" field a UI element? If so, it should be bolded.

igpetrov replied:
Done

christinaausley reviewed (changes requested):
In addition to @MaxTru 's comments on adding links to resources as appropriate, I have a few comments for clarification.

You'll also see a commit labeled `style(formatting): technical review` come through where I've adjusted some of the grammar, syntax, and punctuation according to our style guide. We can't have H1 headers in our docs, so you'll see I've adjusted this as well. Please let me know if you have any questions about this. Thanks!

igpetrov commented:
Notes on the latest changes:
- Addressed comments
- Added documentation related to the OAuth refresh token authorization
- Updated screenshot with updated template

igpetrov commented:
<img width="950" alt="Screenshot 2022-08-30 at 13 24 19" src="https://user-images.githubusercontent.com/108870003/187414284-07d59706-f900-420c-b2c7-afe8785152e2.png">
<img width="690" alt="Screenshot 2022-08-30 at 13 25 22" src="https://user-images.githubusercontent.com/108870003/187414295-d107bac9-40aa-4b23-9daf-dddb5c983c36.png">
<img width="694" alt="Screenshot 2022-08-30 at 13 25 48" src="https://user-images.githubusercontent.com/108870003/187414304-8ff74617-94e6-424c-8332-203545fd197d.png">
<img width="676" alt="Screenshot 2022-08-30 at 13 26 08" src="https://user-images.githubusercontent.com/108870003/187414308-fcdd9bc2-d960-4ef5-aff0-e7abad3889d8.png">
<img width="676" alt="Screenshot 2022-08-30 at 13 26 29" src="https://user-images.githubusercontent.com/108870003/187414311-c70c8ca4-b13c-4e53-b6b0-52fadf7c85cb.png">
<img width="670" alt="Screenshot 2022-08-30 at 13 26 43" src="https://user-images.githubusercontent.com/108870003/187414314-52d6929f-0b4d-4708-bd6e-456e682c79fd.png">
<img width="683" alt="Screenshot 2022-08-30 at 13 26 58" src="https://user-images.githubusercontent.com/108870003/187414318-29b9049e-5cb8-4559-b556-93c1e0bde65a.png">
<img width="662" alt="Screenshot 2022-08-30 at 13 27 08" src="https://user-images.githubusercontent.com/108870003/187414325-2d5aa840-d22a-4edf-8a06-386e506f98b6.png">

christinaausley reviewed (approved):
Updated my branch and committed some technical adjustments as it looks like I didn't push last time. Looks good on my end!

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:
```diff
@@ -9,5 +9,6 @@ This section gives an overview of the **out-of-the-box connectors** available in
 - [REST connector](rest.md) - Make a request to a REST API and use the response in the next steps of your process.
 - [SendGrid connector](sendgrid.md) - Quickly send emails from your BPMN processes.
 - [Slack connector](slack.md) - Send messages to channels or users in your [Slack](https://slack.com) workspace from your BPMN process.
+- [Google Drive connector](googledrive.md) - Create empty folders or documents on [Google Drive](https://www.google.com/drive/) from templates from your BPMN process.
```
I propose to order them alphabetically

igpetrov replied:
Done

MaxTru commented on sidebars.js:
```diff
@@ -310,6 +310,7 @@ module.exports = {
                 "components/integration-framework/connectors/out-of-the-box-connectors/rest",
                 "components/integration-framework/connectors/out-of-the-box-connectors/sendgrid",
                 "components/integration-framework/connectors/out-of-the-box-connectors/slack",
+                "components/integration-framework/connectors/out-of-the-box-connectors/googledrive",
```
I propose to order them alphabetically

igpetrov replied:
Done

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,234 @@
+
+To use a **Google Drive Connector** in your process, either change the type of existing task by clicking on it and using the wrench-shaped **Change type** context menu icon or create a new Connector task by using the **Append Connector** context menu. Follow our [guide on using Connectors](../use-connectors.md) to learn more.
+
+## Make your Google Drive Connector for creating a folder executable
```
To remove redundancy and get away from this long heading, have you considered a structure like this:

```
## Make your Google Drive Connector executable

To make the **Google Drive Connector** executable, fill out the mandatory fields highlighted in red in the properties panel:

![Google Drive Connector new folder example](../img/connectors-googledrive-new-folder-filled.png)

For all oeprations, you need to fill the required credentials in the **Authentication** section. See the [relevant appendix entry](#how-can-i-authenticate-my-connector) to find out more.

You can then decide between two operations: [creating a folder](link) or [creating a file from template](link)

### Create a new folder
...

### Create a file from a template
...
```

igpetrov replied:
Agreed. Updated.

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
````diff
@@ -0,0 +1,234 @@
+Assuming you have created an OAuth client, you should an OAuth Client ID and an OAuth Client secret. Run the following code snippet that prints the refresh token in terminal:
+
+```python
+TBD - prototype code will be added soon
````
Note this is open

igpetrov replied:
Added

MaxTru commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,234 @@
+
+### How do I set additional properties or metadata?
+
+You can set any property from the Google Drive Create API. For example:
```
Can we link to Google Drive Create API explaining which properties exist?

MaxTru reviewed (changes requested):
Please see some small comments

PreetiNirwal commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:
```diff
@@ -6,8 +6,9 @@ description: Take a closer look at the Connectors available in Camunda Platform
-- [REST Connector](rest.md) - Make a request to a REST API and use the response in the next steps of your process.
-- [SendGrid Connector](sendgrid.md) - Quickly send emails from your BPMN processes.
-- [Slack Connector](slack.md) - Send messages to channels or users in your [Slack](https://slack.com) workspace from your BPMN process.
+- [Google Drive connector](googledrive.md) - Create empty folders or documents on [Google Drive](https://www.google.com/drive/) from templates from your BPMN process.
```
Update text to: 
"Create folders or files from a [Google Drive] (https://www.google.com/drive/) template from your BPMN process."

PreetiNirwal replied:
Also can the order of the list be: 
1. REST Connector
2. SendGrid Connector
3. Slack Connector
4. Google Drive Connector
(For our other new connectors coming)
5. AWS Simple Queue Service Connector
6. AWS Lambda Connector

igpetrov replied:
Addressed in a new revision. Will publish shortly.

Regarding connectors order - there was a suggestion from Max to keep them alphabetical.

PreetiNirwal replied:
Ok cool (on the alphabetical) piece!

PreetiNirwal commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,245 @@
+---
+id: googledrive
+title: Google Drive Connector
+description: Create empty folders or documents from templates from your BPMN process.
```
Pls update text to:
"Create folders or files from a Google Drive template from your BPMN process."

igpetrov replied:
Addressed in a new revision. Will publish shortly.

PreetiNirwal commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,245 @@
+
+## Prerequisites
+
+To start working with the **Google Drive Connector**, a relevant OAuth token must be configured and stored as a Secret in the user's cluster. The token must have permission to read and create a folder and/or files from a desired Google Drive instance. Follow the steps from the [appendix](#appendix--faq) to find out more about creating an OAuth token and giving relevant permissions.
```
Pls update text from: "....stored as a Secret in the user's cluster." 
to:  "....stored as a Secret in your cluster."

igpetrov replied:
Addressed in a new revision. Will publish shortly.

PreetiNirwal commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,245 @@
+
+## Create a Google Drive Connector task
+
+Currently, the **Google Drive Connector** supports two types of templates: create a folder and create a file from a template.
```
Pls update text to:
"Currently, the **Google Drive Connector** supports two types of operations: create a folder and create a file from a template."

igpetrov replied:
Addressed in a new revision. Will publish shortly.

PreetiNirwal commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,245 @@
+3. In the **Operation Details** section, set the field **File name** as the desired name of a folder you wish to create. You can use FEEL expressions here.
+4. In the **Operation Details** section, set the field **Template ID** of the desired template.
+5. _(optional)_ In the **Operation Details** section, set the field **Parent folder ID** to the desired parent, inside which a new file will be created. Keep in mind that if not specified, a new folder will be created in the Google Drive root folder of a user who owns the OAuth token.
+6. In the **Operation Details** section, set the field **Template variables** as desired variables that will be applied to the template. The template variables a compatible with the Google Docs Requests API. This property requires FEEL input.
```
Pls update text from: "The template variables a compatible with the Google Docs Requests API."
to: "The template variables are compatible with the Google Docs Requests API."

igpetrov replied:
Addressed in a new revision. Will publish shortly.

PreetiNirwal commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,245 @@
+
+### What Google API does the Google Drive Connector use to create a folder?
+
+The **Google Drive Connector** uses Google Drive [`Files:Create`](https://developers.google.com/drive/api/v3/reference/files/create) API endpoint.
```
Pls update text to: "The **Google Drive Connector** uses the Google Drive...."

igpetrov replied:
Addressed in a new revision. Will publish shortly.

PreetiNirwal commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,245 @@
+
+### What Google API does the Google Drive Connector use to create a file from template?
+
+The **Google Drive Connector** uses Google Drive [`Files:Copy`](https://developers.google.com/drive/api/v3/reference/files/copy) API endpoint to copy an original template. Afterwards, the **Google Drive Connector** utilizes Google Docs [Merge](https://developers.google.com/docs/api/how-tos/merge) approach via [`Documents:BatchUpdate`](https://developers.google.com/docs/api/reference/rest/v1/docum ...
```
Pls update first sentence to: "The Google Drive Connector uses the Google Drive...."

igpetrov replied:
Addressed in a new revision. Will publish shortly.

PreetiNirwal commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md:
```diff
@@ -0,0 +1,245 @@
+
+### Can you show me an example of a valid template?
+
+The valid template looks like the one on a screenshot.
```
Pls update text to: "A valid template looks like the following screenshot."

igpetrov replied:
Addressed in a new revision. Will publish shortly.

igpetrov commented:
Published new new revision

igpetrov commented:
@PreetiNirwal, updated wording and screenshots.

PreetiNirwal reviewed (dismissed):
Thanks for the updates! 
Looks great!

PreetiNirwal commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:
```diff
@@ -6,8 +6,9 @@ description: Take a closer look at the Connectors available in Camunda Platform
-- [REST Connector](rest.md) - Make a request to a REST API and use the response in the next steps of your process.
-- [SendGrid Connector](sendgrid.md) - Quickly send emails from your BPMN processes.
-- [Slack Connector](slack.md) - Send messages to channels or users in your [Slack](https://slack.com) workspace from your BPMN process.
+- [Google Drive connector](googledrive.md) - Create folders or files from a [Google Drive] (https://www.google.com/drive/) template from your BPMN process.
```
```suggestion
- [Google Drive Connector](googledrive.md) - Create folders or files from a [Google Drive] (https://www.google.com/drive/) template from your BPMN process.
```

PreetiNirwal commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:
```diff
@@ -6,8 +6,9 @@ description: Take a closer look at the Connectors available in Camunda Platform
-- [SendGrid Connector](sendgrid.md) - Quickly send emails from your BPMN processes.
-- [Slack Connector](slack.md) - Send messages to channels or users in your [Slack](https://slack.com) workspace from your BPMN process.
+- [Google Drive connector](googledrive.md) - Create folders or files from a [Google Drive] (https://www.google.com/drive/) template from your BPMN process.
+- [REST connector](rest.md) - Make a request to a REST API and use the response in the next steps of your process.
```
```suggestion
- [REST Connector](rest.md) - Make a request to a REST API and use the response in the next steps of your process.
```

PreetiNirwal commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:
```diff
@@ -6,8 +6,9 @@ description: Take a closer look at the Connectors available in Camunda Platform
-- [Slack Connector](slack.md) - Send messages to channels or users in your [Slack](https://slack.com) workspace from your BPMN process.
+- [Google Drive connector](googledrive.md) - Create folders or files from a [Google Drive] (https://www.google.com/drive/) template from your BPMN process.
+- [REST connector](rest.md) - Make a request to a REST API and use the response in the next steps of your process.
+- [SendGrid connector](sendgrid.md) - Quickly send emails from your BPMN processes.
```
```suggestion
- [SendGrid Connector](sendgrid.md) - Quickly send emails from your BPMN processes.
```

PreetiNirwal commented on docs/components/integration-framework/connectors/out-of-the-box-connectors/available-connectors-overview.md:
```diff
@@ -6,8 +6,9 @@ description: Take a closer look at the Connectors available in Camunda Platform
+- [Google Drive connector](googledrive.md) - Create folders or files from a [Google Drive] (https://www.google.com/drive/) template from your BPMN process.
+- [REST connector](rest.md) - Make a request to a REST API and use the response in the next steps of your process.
+- [SendGrid connector](sendgrid.md) - Quickly send emails from your BPMN processes.
+- [Slack connector](slack.md) - Send messages to channels or users in your [Slack](https://slack.com) workspace from your BPMN process.
```
```suggestion
- [Slack Connector](slack.md) - Send messages to channels or users in your [Slack](https://slack.com) workspace from your BPMN process.
```

PreetiNirwal reviewed (approved):
Minor update only, we are capitalising Connectors.

christinaausley reviewed (approved):
Technical adjustments in latest commit.
